doc/core/moving-to-c++.md

```diff
@@ -0,0 +1,54 @@
+# Moving gRPC core to C++
+
+October 2017
+
+ctiller, markdroth, vjpai
+
+## Background and Goal
+
+gRPC core was originally written in C89 for several reasons (possibility of
```

Comment:
Might be useful to add a link to https://github.com/grpc/proposal/blob/master/L6-allow-c%2B%2B-in-grpc-core.md for context.  Although it looks like the plan we discussed on Friday is somewhat different from the one described in the gRFC.

Reply:
Sounds good, I did that.